src/web/js/error-ui.js

```diff
@@ -55,6 +55,46 @@
       }
 
       // HELPERS
+      // NOTE: MUST BE CALLED WHEN RUNNING ON runtime's STACK
+      function renderSimpleReason(err) {
+        var ret = Q.defer();
+        runtime.runThunk(function() { 
+          return get(err.exn || err, "render-reason").app(); 
```

Comment:
> I'm worried about these short little hacks littering the code and leading to accidental undefined values.
Having a handful of shapes that errors can come is the root of the issue here (and the result of most of the verbosity of `error-ui.js`; some of the structures that file accommodate before the error overhaul weren't used anywhere, even then). A consistent shape for error values would simplify things in a more maintainable way.